### PR TITLE
Document recording cassettes, make it possible to re-record for subset of tests

### PR DESCRIPTION
**Pull request by bkabrda**

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -4,6 +4,7 @@ WEBSITE_REPO=github.com/hashicorp/terraform-website
 PKG_NAME=datadog
 DIR=~/.terraform.d/plugins
 GO_CLIENT_VERSION=master
+RUNTESTS?=""
```
Comment (jirikuncar) [inline GNUmakefile]: you should be able to do this with `$(TESTARGS)`

File: GNUmakefile [hunk 2]
```diff
@@ -21,14 +22,14 @@ test: fmtcheck
 	go test -i $(TEST) || exit 1
 	echo $(TEST) | \
 		xargs -t -n4 go test $(TESTARGS) -timeout=30s -parallel=4
-	DATADOG_API_KEY=fake DATADOG_APP_KEY=fake RECORD=false TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout=120s
+	DATADOG_API_KEY=fake DATADOG_APP_KEY=fake RECORD=false TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout=120s -run=$(RUNTESTS)
```
Comment (jirikuncar) [inline GNUmakefile]:
```suggestion
	DATADOG_API_KEY=fake DATADOG_APP_KEY=fake RECORD=false TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout=120s
```

File: GNUmakefile [hunk 3]
```diff
@@ -21,14 +22,14 @@ test: fmtcheck
...
 
 testacc: fmtcheck
-	TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout 120m
+	TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout 120m -run=$(RUNTESTS)
```
Comment (jirikuncar) [inline GNUmakefile]:
```suggestion
	TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout 120m
```

File: GNUmakefile [hunk 4]
```diff
@@ -21,14 +22,14 @@ test: fmtcheck
...
 
 cassettes: fmtcheck
 	rm datadog/cassettes/*
-	RECORD=true TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout 120m
+	RECORD=true TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout 120m -run=$(RUNTESTS)
```
Comment (jirikuncar) [inline GNUmakefile]:
```suggestion
	RECORD=true TF_ACC=1 go test $(TEST) -v $(TESTARGS) -timeout 120m
```

File: README.md [hunk 5]
```diff
@@ -58,9 +58,24 @@ In order to test the provider, you can simply run `make test`.
...
+(cassettes) stored in `datadog/cassettes/`. When tests are modified, the cassettes need
+to be re-recorded.
+
+*Note:* Recording cassettes creates/updates/destroys real resources. Never run this on
+a production Datadog organization.
+
+In order to re-record all cassettes you need to have `DATADOG_API_KEY` and `DATADOG_APP_KEY`
+for your testing organization in your environment. With that, run `make cassettes`. Do note
+that this would regenerate all cassettes and thus take a very long time; if you only need to
+re-record cassettes for one or two tests, you can run `make cassettes RUNTESTS=XXX` - this
```
Comment (jirikuncar) [inline README.md]:
```suggestion
re-record cassettes for one or two tests, you can run `make cassettes TESTARGS ="-run XXX"` - this
```

Comment (jirikuncar) [reply README.md]: Have you check it `make -e TESTARGS="-run XXX" cassettes` works? There are some differences between GNUMake and BSD make.

Comment (bkabrda) [reply README.md]: Indeed it does seem to work.

File: README.md [hunk 6]
```diff
@@ -58,9 +58,32 @@ In order to test the provider, you can simply run `make test`.
...
+(cassettes) stored in `datadog/cassettes/`. When tests are modified, the cassettes need
+to be re-recorded.
+
+In order to make tests cassette friendly, it's necessary to ensure that resources always get
+manipulated in a predictable order. When creating a testing Terraform config that defines multiple
+resources at the same time, you need to set inter-resource dependencies (using `depends_on`)
+in such a way that there is only one way for Terraform to manipulate them. For example, given
+resources A, B and C in the same config string, you can achieve this by making A depend on B
+and B depend on C. See [PR #442](https://github.com/terraform-providers/terraform-provider-datadog/pull/442)
+for an example of this.
```
Comment (jirikuncar) [inline README.md]: 👍

File: README.md [hunk 7]
```diff
@@ -58,9 +58,32 @@ In order to test the provider, you can simply run `make test`.
...
+
+*Note:* Recording cassettes creates/updates/destroys real resources. Never run this on
+a production Datadog organization.
+
```
Comment (jirikuncar) [inline README.md]: It looks like a duplicate of lines 85-86
```suggestion
```

Comment (bkabrda) [reply README.md]: It's not a duplicate. This line is about cassettes, lines 85-86 are about integration tests. These are 2 distinct commands that both would use keys and do various changes on the DD org, so let's include explicit warning for both of them.

**Comment by bkabrda** [comment]
Thanks for the review, merging.

**Pull request merged**